plugins/catalog-backend-module-openapi/src/lib/bundle.ts

```diff
@@ -57,7 +57,8 @@ export async function bundleFileWithRefs(
       return protocol === 'http' || protocol === 'https';
     },
     read: async ref => {
-      const url = resolveUrl(ref.url, baseUrl);
+      const file = ref.url.split('/');
+      const url = resolveUrl(file[file.length - 1], baseUrl);
```

Comment:
This will only work for the last part of a file name, we'd want this to resolve for all files. I think this code was fine before, I believe the issue is on line 49 with the `path.relative('.', ...)`, in your example the ref doesn't have a protocol, so it should hit the fileUrlReaderResolver.

Reply:
From the tests I ran locally, the ref is resolved as "https://...", and this causes the error described in the issue 27280.

Reply:
gotcha, thanks for figuring this out! can we add a test here as well?

Reply:
My server exposes the files hw.yaml and commons.yaml. 
hw.yaml
`openapi: 3.0.3
info:
  title: Swagger Petstore - OpenAPI 3.0
  version: 1.0.11
tags:
  - name: pet
    description: Everything about your Pets
    externalDocs:
      description: Find out more
      url: http://swagger.io
paths:
  /pet:
    put:
      tags:
        - pet
      summary: Update an existing pet
      description: Update an existing pet by Id
      operationId: updatePet
      requestBody:
        description: Update an existent pet in the store
        content:
          application/json:
            schema:
              $ref: 'commons.yaml#/components/schemas/Pet'
          application/xml:
            schema:
              $ref: 'commons.yaml#/components/schemas/Pet'
          application/x-www-form-urlencoded:
            schema:
              $ref: 'commons.yaml#/components/schemas/Pet'
        required: true
      responses:
        '200':
          description: Successful operation
          content:
            application/json:
              schema:
                $ref: 'commons.yaml#/components/schemas/Pet'          
            application/xml:
              schema:
                $ref: 'commons.yaml#/components/schemas/Pet'
        '400':
          description: Invalid ID supplied
        '404':
          description: Pet not found
        '422':
          description: Validation exception`
commons.yaml
`openapi: "3.0.0"
info:
  version: 1.0.0
  title: commons
  description: Commons library
paths: {}

components:
  schemas:
    Pet:
      required:
        - name
      type: object
      properties:
        id:
          type: integer
          format: int64
          example: 10
        name:
          type: string
          example: doggie
      xml:
        name: pet`

I added the following comments in the code:
![image](https://github.com/user-attachments/assets/c7a2c6cb-2bfe-4944-bb59-480e2fdaceff)

 When I import the API into Backstage, I get:
![image](https://github.com/user-attachments/assets/33f6cde5-73c9-48b2-846d-5eb67105f6ce)